demo/ContainerInteractor.swift

```diff
@@ -214,21 +215,20 @@ extension ContainerInteractor: VisionManagerDelegate {
     }
     
     func visionManager(_ visionManager: VisionManager, didUpdateWorldDescription worldDescription: WorldDescription?) {
-        guard
-            case .distanceToObject = currentScreen,
-            let roadDescription = visionManager.roadDescription,
-            let car = worldDescription?.objects.first,
-            roadDescription.currentLane < roadDescription.lanes.count,
-            car.objectType == .car
-        else {
+        guard case .distanceToObject = currentScreen, let worldDescription = worldDescription else {
+            presenter.present(collisionObjectFrame: nil, canvasSize: visionManager.frameSize)
             presenter.present(distanceToCar: nil, canvasSize: visionManager.frameSize)
             return
         }
         
-        let width = roadDescription.lanes[roadDescription.currentLane].width
-
-        let carLeftPosition = visionManager.worldToPixel(worldCoordinate: WorldCoordinate(x: car.worldCoordinate.x - (width / 2), y: car.worldCoordinate.y, z: car.worldCoordinate.z))
-        let carRightPosition = visionManager.worldToPixel(worldCoordinate: WorldCoordinate(x: car.worldCoordinate.x + (width / 2), y: car.worldCoordinate.y, z: car.worldCoordinate.z))
+        guard let car = worldDescription.collisionObjects.first.map({ $0.object }), car.objectType == .car else {
```

Comment:
Couple of issues here.
- If `worldDescription != nil` but `collisionObjects` is empty, collision state won't be cleared from the screen.
- As I can see we're drawing distance to object even on collision state although these states are mutually exclusive by design.
- Collision state `.warning` is not handled. Question to @rokkeds how to do it.

I can see the following algorithm here:

```suggestion
        guard 
            case .distanceToObject = currentScreen, 
            let worldDescription = worldDescription,
            let car = worldDescription.collisionObjects.first, 
            car.detection.objectType == .car 
        else {
            // no target object so cleanup everything and return
        }
        
        if car.state == .notTriggered {
            // cleanup collision state
            // draw distance to object
        } else {
            // cleanup distance to object
            // draw collision state
        }
            
```

It may be resolved with an `if`s like above or some kind of state for this screen like
```swift
enum DistanceToObjectScreenState {
    case none
    case calibration(progress: Int)
    case distance(distance: DistanceToCar, canvasSize: CGSize)
    case collision(collisionState: CollisionState, canvasSize: CGSize)
}
```
and presenting the state on `presenter`. This way the `presenter` will resolve showing / hiding logic and states would be strictly exclusive.